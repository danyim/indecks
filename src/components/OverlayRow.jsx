import PropTypes from 'prop-types'
import React from 'react'
import styles from '../styles/components/Overlay.styl'

const propTypes = {
  className: PropTypes.string,
  children: PropTypes.node
}

const defaultProps = {
  className: '',
  children: null
}

const OverlayRow = (props) => {
  // Prepends the hover class to the original class on the child
  const modifiedChildren = React.Children.map(props.children, (c) => {
    return React.cloneElement(c, {
      className: `${styles['hover-button']}${c.props.className ? ' ' + c.props.className : ''}`
    })
  })

  return (
    <div className={`${props.className} ${styles['hover-actions']}`}>
      {modifiedChildren}
    </div>
  )
}

OverlayRow.propTypes = propTypes
OverlayRow.defaultProps = defaultProps

export default OverlayRow
