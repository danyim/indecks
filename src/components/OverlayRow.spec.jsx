import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import OverlayRow from './OverlayRow'

const defaultProps = {
  className: '',
  children: null
}

function setup(props = defaultProps) {
  const wrapper = shallow(<OverlayRow {...props} />)

  return {
    props,
    wrapper
  }
}

describe('OverlayRow', () => {
  it('should render self and subcomponents', () => {
    const { wrapper } = setup()
    expect(wrapper.find('div').hasClass('hover-actions')).toBe(true)

    const tree = renderer.create(<OverlayRow {...defaultProps} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should apply a class to all its children', () => {
    const children = <a className="test">Test</a>
    const { wrapper } = setup({
      ...defaultProps,
      children
    })

    expect(wrapper.find('a').exists()).toBe(true)
    expect(wrapper.find('a').hasClass('test')).toBe(true)
    expect(wrapper.find('a').hasClass('hover-button')).toBe(true)
  })
})
