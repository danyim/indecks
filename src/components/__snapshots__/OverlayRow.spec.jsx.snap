// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverlayRow should render self and subcomponents 1`] = `
<div
  className=" hover-actions"
/>
`;
